starmap_client/utils.py

```diff
@@ -0,0 +1,25 @@
+from typing import Any, Dict
+
+
+def assert_is_dict(data: Any) -> None:
+    """Ensure the incoming data is a dictionary, raises ``ValueError`` if not."""
+    if not isinstance(data, dict):
+        raise ValueError(f"Expected dictionary, got {type(data)}: {data}")
+
+
+def dict_merge(a: Dict[str, Any], b: Dict[str, Any]) -> Dict[str, Any]:
+    """Return a new dictionary with the combination of A and B.
+
+    Args:
+        a (dict):
+            The left dictionary to be combined
+        b (dict):
+            The right dictionary to be combined. It will override the same keys from A.
+    Returns:
+        dict: A new dictionary with combination of A and B.
+    """
+    c = {}
```

Comment:
instead of c = {} you could do 'return a | b' . Just a suggestion

Reply:
I've tried that but IIUC `a | b` requires the keys to not overlap.

In this method we're considering that one or more keys may be overlapping so `b` would be determinant over `a` if that happens.

Reply:
Ok, let me try to adjust it 😄

Reply:
I've changed it, thanks for the suggestion!

Reply:
Oh.. It seems to have broken `py38` 😞

Reply:
I'm not sure whether we can drop it since `pub` might still use this version...

@lslebodn thoughts?

Reply:
@lslebodn JFTR: https://github.com/release-engineering/starmap-client/compare/b892f06ab1d4c91106206de582d4051fd6cb5284..c8c6ff4805a15e4b0491290a1bc1e52b284e76f0

Reply:
I'm sorry! I forgot that pub still has py38. From what I've read the | operator was added py39+

Reply:
@jajreidy Added on 3d49924361e696485e8b7226cde895886b69def4

@lslebodn thanks a lot for allowing me to merge #47